Foundation/NSURLSession/http/EasyHandle.swift

```diff
@@ -240,9 +240,18 @@ extension _EasyHandle {
         try! CFURLSession_easy_setopt_int64(rawHandle, CFURLSessionOptionINFILESIZE_LARGE, length).asError()
     }
 
-   func set(timeout value: Int) {
+    func set(timeout value: Int) {
        try! CFURLSession_easy_setopt_long(rawHandle, CFURLSessionOptionTIMEOUT, value).asError()
-   }
+    }
+
+    func getTimetaken() -> Double {
+        let timeTaken = UnsafeMutablePointer<Double>.allocate(capacity: 8)
```

Comment:
This is not correct. You are allocating 8 instances of type `Double`. The body of this function should be:

```swift
var timeSpent = Double()
CFURLSession_easy_getinfo_double(rawHandle, CFURLSessionInfoTOTAL_TIME,  &timeSpent)
return timeSpent
```

Since this is used only for subtracting from the timeout interval, it might be worthwhile to return `timeTaken / 1000` and to rename this function `getTimeoutIntervalSpent()`.